openstf/pipeline/train_model_sklearn.py

```diff
@@ -29,59 +26,69 @@
-#         logger.warning("No old model found, train new model")
-
-#     # Check old model age and continue yes/no
-#     if (old_model_age < MAXIMUM_MODEL_AGE) and check_old_model_age:
-#         logger.warning(
-#             "Old model is younger than {MAXIMUM_MODEL_AGE} days, skip training"
-#         )
-#         return
-
-#     # train model
-#     try:
-#         model, report = train_model_pipeline_core(pj, input_data, old_model)
-#     except RuntimeError as e:
-#         return
-
-#     # save model
-#     serializer.save_model(model, pid=pj["id"])
-#     # save figures
-#     report.save_figures(save_path=save_figures_folder)
+def train_model_pipeline(
+    pj: dict,
+    input_data: pd.DataFrame,
+    check_old_model_age: bool,
+    trained_models_folder: Union[str, Path],
+    save_figures_folder: Union[str, Path],
+) -> None:
+    """Midle level pipeline that takes care of all persistent storage dependencies
+
+    Args:
+        pj (dict): Prediction job
+        input_data (pd.DataFrame): Raw training input data
+        check_old_model_age (bool): Check if training should be skipped because the model is too young
+        trained_models_folder (Path): Path where trained models are stored
+        save_figures_folder (Path): path were reports (mostly figures) about the training procces are stored
+
+    Returns:
+        None
+
+    """
+    # Intitialize logger and serializer
+    logger = structlog.get_logger(__name__)
+    serializer = PersistentStorageSerializer(trained_models_folder)
+
+    # Get old model and age
+    try:
+        old_model = serializer.load_model(pid=pj["id"])
+        old_model_age = (
+            old_model.age
+        )  # Age attribute is openstf specific and is added by the serializer
+    except FileNotFoundError:
+        old_model = None
+        old_model_age = float("inf")
+        logger.warning("No old model found, training new model")
+
+    # Check old model age and continue yes/no
+    if (old_model_age < MAXIMUM_MODEL_AGE) and check_old_model_age:
+        logger.warning(
+            f"Old model is younger than {MAXIMUM_MODEL_AGE} days, skip training"
+        )
+        return
+
+    # Train model with core pipeline
+    try:
+        model, report = train_model_pipeline_core(pj, input_data, old_model)
+    except RuntimeError as e:
+        logger.error(f"Exception occured during training: {e}")
+        return
+
+    # Save model
+    serializer.save_model(model, pid=pj["id"])
+
+    # Save reports/figures
+    report.save_figures(save_path=save_figures_folder)
 
 
 def train_model_pipeline_core(
     pj: dict,
     input_data: pd.DataFrame,
     old_model: RegressorMixin = None,
-    horizons: List[float] = DEFAULT_TRAIN_HORIZONS,
-) -> Tuple[RegressorMixin, Report]:
-    """Run the train model pipeline.
+    horizons: List[float] = None,
+) -> tuple[RegressorMixin, Report]:
+    """Run the train model core pipeline. This pipeline has no database or persisitent storage dependencies.
```

Comment:
Same as the other docstrings. You probably know. But if in doubt there are good resources:

[Python Google docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
[PEP-257](https://www.python.org/dev/peps/pep-0257)